### PR TITLE
Internal: Conversion Center - Variation 7 and refactoring

### PR DESCRIPTION
**Pull request by nicoladj77**
## PR Checklist
- [x] The commit message follows our guidelines:  https://github.com/elementor/elementor/blob/master/.github/CONTRIBUTING.md

## PR Type
What kind of change does this PR introduce?

- [ ] Bugfix
- [x] Feature
- [ ] Code style update (formatting, local variables)
- [x] Refactoring (no functional changes, no api changes)
- [ ] Build related changes
- [ ] CI related changes
- [ ] Documentation content changes
- [ ] Other... Please describe:

## Summary

This PR can be summarized in the following changelog entry:

* Added controls for variation 7
* Refactored Code

## Description
An explanation of what is done in this PR

* Added controls for variation 7
* Refactored Code

## Test instructions
This PR can be tested by following these steps:

*

## Quality assurance

- [x] I have tested this code to the best of my abilities
- [ ] I have added unittests to verify the code works as intended
- [ ] Docs have been added / updated (for bug fixes / features)

Fixes #

### CODE REVIEW

File: modules/conversion-center/base/widget-link-in-bio-base.php [hunk 1]
```diff
@@ -639,906 +650,970 @@ protected function get_icon_title_field(): string {
...
+				'placeholder' => esc_html__( 'Description', 'elementor' ),
+				'default' => $config['content']['bio_section']['description']['default'],
+			]
+		);
+
+		$this->end_controls_section();
+	}
 
+	protected function add_identity_section(): void {
+		$config = static::get_configuration();
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-link-in-bio-base.php:901]:
```suggestion
		$config = static::get_configuration();

```
Add empty line

File: modules/conversion-center/base/widget-link-in-bio-base.php [hunk 2]
```diff
@@ -639,906 +650,970 @@ protected function get_icon_title_field(): string {
...
+			);
 		}
-		$this->add_group_control(
-			Group_Control_Background::get_type(),
+
+		$this->end_controls_section();
+	}
+
+	protected function add_style_cta_section(): void {
+		$config = static::get_configuration();
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-link-in-bio-base.php:972]:
```suggestion
		$config = static::get_configuration();

```
Add empty line

File: modules/conversion-center/base/widget-link-in-bio-base.php [hunk 3]
```diff
@@ -639,906 +650,970 @@ protected function get_icon_title_field(): string {
...
+					],
+				]
+			);
+		}
+
+		$this->end_controls_section();
+	}
+
+	protected function add_style_identity_controls(): void {
+		$config = static::get_configuration();
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-link-in-bio-base.php:1151]:
```suggestion
		$config = static::get_configuration();

```
Add empty line

File: modules/conversion-center/base/widget-link-in-bio-base.php [hunk 4]
```diff
@@ -639,906 +650,970 @@ protected function get_icon_title_field(): string {
...
 		);
 
+		$condition = [];
+		if ( $config['content']['identity_section']['identity_image_style'] ) {
+			$condition = [
+				'identity_image_style' => 'profile',
+			];
+		}
+
+		// Profile image style controls
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-link-in-bio-base.php:1167]: Please remove this comment

File: modules/conversion-center/base/widget-link-in-bio-base.php [hunk 5]
```diff
@@ -639,906 +650,970 @@ protected function get_icon_title_field(): string {
...
+			'background' => [
+				'default' => 'classic',
+			],
+		];
+
+		$config = static::get_configuration();
+		if ( $config['style']['border_section']['field_options'] ) {
+			$bg_image_field_options = array_merge( $bg_image_field_options, $config['style']['border_section']['field_options'] );
+		}
+		$this->add_group_control(
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-link-in-bio-base.php:1507]:
```suggestion
		$config = static::get_configuration();

		if ( $config['style']['border_section']['field_options'] ) {
			$bg_image_field_options = array_merge( $bg_image_field_options, $config['style']['border_section']['field_options'] );
		}

		$this->add_group_control(
```

File: modules/conversion-center/traits/conversion-center-controls-trait.php [hunk 6]
```diff
@@ -3,6 +3,8 @@
 namespace Elementor\Modules\ConversionCenter\Traits;
 
 use Elementor\Controls_Manager;
+use Elementor\Modules\ConversionCenter\Classes\Providers\Social_Network_Provider;
+use Elementor\Utils;
```
Comment (louiswol94) [inline modules/conversion-center/traits/conversion-center-controls-trait.php:7]:
```suggestion
```
Do we need these imports?

File: modules/conversion-center/base/widget-link-in-bio-base.php [hunk 7]
```diff
@@ -390,17 +401,22 @@ protected function add_icons_controls(): void {
...
-			[
-				'type' => Controls_Manager::ALERT,
-				'alert_type' => 'info',
-				'content' => sprintf(
-					__( 'Add up to <b>%d</b> icons', 'elementor' ),
-					5
-				),
-			]
-		);
+		do_action( 'elementor/conversion-center/content/controls/icons/before', $this );
```
Comment (louiswol94) [inline modules/conversion-center/base/widget-link-in-bio-base.php:404]: Why do we need this hook? Can we use `$this->start_injection()` instead?

**Review by louiswol94 — CHANGES_REQUESTED** [review]

File: modules/conversion-center/base/widget-link-in-bio-base.php (hunk 7 above)
Comment (nicoladj77) [reply modules/conversion-center/base/widget-link-in-bio-base.php:404]: @louiswol94 not sure how `$this->start_injection()` would work.  I tried looking at examples but not 100% clear.
This is used to inject a control that could be here or could be in a different control section based on the variation, so I put the control in the trait and used an action.

Comment (louiswol94) [reply modules/conversion-center/base/widget-link-in-bio-base.php:404]: @nicoladj77 Sorry, not sure I understand. If the `do_action()` is here, the control will always be rendered here. How can it be here, or in a different control section?

Comment (louiswol94) [reply modules/conversion-center/base/widget-link-in-bio-base.php:404]: @nicoladj77 Maybe this is an example?
https://github.com/elementor/elementor-pro/blob/main/modules/carousel/widgets/media-carousel.php#L131

Comment (nicoladj77) [reply modules/conversion-center/base/widget-link-in-bio-base.php:404]: @louiswol94 ok clear now.

**Review by louiswol94 — APPROVED** [review]

**Pull request merged**